manager/scheduler/scheduler_test.go

```diff
@@ -2168,6 +2191,129 @@ func TestSchedulerCompatiblePlatform(t *testing.T) {
 	assert.Regexp(t, assignment4.NodeID, "(node1|node2)")
 }
 
+func TestSchedulerUnassignedMap(t *testing.T) {
+	ctx := context.Background()
+	// create a service and a task with OS constraint that is not met
+	task1 := &api.Task{
+		ID:           "id1",
+		ServiceID:    "serviceID1",
+		DesiredState: api.TaskStateRunning,
+		ServiceAnnotations: api.Annotations{
+			Name: "name1",
+		},
+		Status: api.TaskStatus{
+			State: api.TaskStatePending,
+		},
+		Spec: api.TaskSpec{
+			Placement: &api.Placement{
+				Platforms: []*api.Platform{
+					{
+						Architecture: "amd64",
+						OS:           "windows",
+					},
+				},
+			},
+		},
+	}
+
+	task2 := &api.Task{
+		ID:           "id2",
+		ServiceID:    "serviceID2",
+		DesiredState: api.TaskStateRunning,
+		ServiceAnnotations: api.Annotations{
+			Name: "name1",
+		},
+		Status: api.TaskStatus{
+			State: api.TaskStatePending,
+		},
+		Spec: api.TaskSpec{
+			Placement: &api.Placement{
+				Platforms: []*api.Platform{
+					{
+						Architecture: "amd64",
+						OS:           "linux",
+					},
+				},
+			},
+		},
+	}
+
+	node1 := &api.Node{
+		ID: "node1",
+		Spec: api.NodeSpec{
+			Annotations: api.Annotations{
+				Name: "node1",
+			},
+		},
+		Status: api.NodeStatus{
+			State: api.NodeStatus_READY,
+		},
+		Description: &api.NodeDescription{
+			Platform: &api.Platform{
+				Architecture: "x86_64",
+				OS:           "linux",
+			},
+		},
+	}
+
+	service1 := &api.Service{
+		ID: "serviceID1",
+	}
+
+	service2 := &api.Service{
+		ID: "serviceID2",
+	}
+
+	s := store.NewMemoryStore(nil)
+	assert.NotNil(t, s)
+	defer s.Close()
+
+	err := s.Update(func(tx store.Tx) error {
+		// Add initial task, service and nodes to the store
+		assert.NoError(t, store.CreateService(tx, service1))
+		assert.NoError(t, store.CreateTask(tx, task1))
+		assert.NoError(t, store.CreateNode(tx, node1))
+		return nil
+	})
+	assert.NoError(t, err)
+
+	scheduler := New(s)
+
+	watch, cancel := state.Watch(s.WatchQueue(), api.EventUpdateTask{})
+	defer cancel()
+
+	go func() {
+		assert.NoError(t, scheduler.Run(ctx))
+	}()
+	defer scheduler.Stop()
+
+	failure := watchAssignmentFailure(t, watch)
+	assert.Equal(t, "no suitable node (unsupported platform on 1 node)", failure.Status.Err)
+
+	// task1 is in the unassigned map
+	assert.Contains(t, scheduler.unassignedTasks, task1.ID)
+
+	// delete the service of an unassigned task
+	err = s.Update(func(tx store.Tx) error {
+		assert.NoError(t, store.DeleteService(tx, service1.ID))
```

Comment:
It might be much simpler test by directly calling tick()

Reply:
was testing the full stack call and let the scheduler does its work, but in this case, make sure to limit the change to unassigned map and the tick() call